lib/shopify_cli/constants.rb

```diff
@@ -38,7 +38,7 @@ module EnvironmentVariables
 
       # When true the CLI points to spin instances of services
       SPIN = "SPIN"
-      INFER_SPIN = "INFER_SPIN"
```

Comment:
Inference should be the default. This covers the vast majority of cases where users are trying to push up a script to their active dev environment, which is typically the most recent spin environment they've created.